apis/v1alpha1/work_group.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WorkGroupSpec defines the desired state of WorkGroup.
+//
+// A workgroup, which contains a name, description, creation time, state, and
+// other configuration, listed under WorkGroup$Configuration. Each workgroup
+// enables you to isolate queries for you or your group of users from other
+// queries in the same account, to configure the query results location and
+// the encryption configuration (known as workgroup settings), to enable sending
+// query metrics to Amazon CloudWatch, and to establish per-query data usage
+// control limits for all queries in a workgroup. The workgroup settings override
+// is specified in EnforceWorkGroupConfiguration (true/false) in the WorkGroupConfiguration.
+// See WorkGroupConfiguration$EnforceWorkGroupConfiguration.
+type WorkGroupSpec struct {
+
+	// Contains configuration information for creating an Athena SQL workgroup or
+	// Spark enabled Athena workgroup. Athena SQL workgroup configuration includes
+	// the location in Amazon S3 where query and calculation results are stored,
+	// the encryption configuration, if any, used for encrypting query results,
+	// whether the Amazon CloudWatch Metrics are enabled for the workgroup, the
+	// limit for the amount of bytes scanned (cutoff) per query, if it is specified,
+	// and whether workgroup's settings (specified with EnforceWorkGroupConfiguration)
+	// in the WorkGroupConfiguration override client-side settings. See WorkGroupConfiguration$EnforceWorkGroupConfiguration.
+	Configuration *WorkGroupConfiguration `json:"configuration,omitempty"`
+	// The workgroup description.
+	Description *string `json:"description,omitempty"`
+	// The workgroup name.
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+	// A list of comma separated tags to add to the workgroup that is created.
+	Tags []*Tag `json:"tags,omitempty"`
```

Comment:
Can we also support tag updates?

Reply:
Can we have it in this PR @TiberiuGC or are planning on doing it seperately?

Reply:
is this done?